chia/wallet/wallet_node.py

```diff
@@ -257,7 +261,11 @@ async def _start(
         )
 
         assert self._wallet_state_manager is not None
-
+        if self._wallet_state_manager.blockchain.synced_weight_proof is not None:
+            weight_proof = self._wallet_state_manager.blockchain.synced_weight_proof
+            success, _, records = await self._weight_proof_handler.validate_weight_proof(weight_proof, True)
```

Comment:
why do we validate it if it's already in synced_weight_proof, don't we already know that it's valid?

Reply:
most of the validation is skipped we do this to get the records from the proof (we don't cache currently)
this happens only on a restart and replaces the initialization we used to do in wallet_blockchain.py